Compiling mapserver 8.0.0 error libmapserver.so.8.0.0: undefined reference to `msWFSIsLayerSupported'

Hi

We currently use the mapserver version 7.6.4 and we are not having any problems with the compilation, but on the mapserver 8.0.0 it is returns an error.

I cloned the project and ran the command below:
```
cmake ../ \
                -DWITH_KML=OFF \
                -DWITH_SOS=OFF \
                -DWITH_WMS=ON \
                -DWITH_FRIBIDI=ON \
                -DWITH_HARFBUZZ=ON \
                -DWITH_ICONV=ON \
                -DWITH_CAIRO=ON \
                -DWITH_SVGCAIRO=OFF \
                -DWITH_RSVG=OFF \
                -DWITH_MYSQL=OFF \
                -DWITH_FCGI=ON \
                -DWITH_GEOS=ON \
                -DWITH_CURL=OFF \
                -DWITH_CLIENT_WMS=OFF \
                -DWITH_WCS=OFF \
                -DWITH_CLIENT_WFS=OFF \
                -DWITH_WFS=OFF \
                -DWITH_LIBXML2=ON \
                -DWITH_THREAD_SAFETY=OFF \
                -DWITH_GIF=ON \
                -DWITH_PYTHON=OFF \
                -DWITH_PHP=OFF \
                -DWITH_PERL=OFF \
                -DWITH_RUBY=OFF \
                -DWITH_JAVA=OFF \
                -DWITH_CSHARP=OFF \
                -DWITH_ORACLESPATIAL=OFF \
                -DWITH_ORACLE_PLUGIN=OFF \
                -DWITH_MSSQL2008=OFF \
                -DWITH_EXEMPI=ON \
                -DWITH_XMLMAPFILE=ON \
                -DWITH_V8=OFF \
                -DBUILD_STATIC=OFF \
                -DLINK_STATIC_LIBMAPSERVER=OFF \
                -DWITH_APACHE_MODULE=OFF \
                -DWITH_GENERIC_NINT=OFF \
                -DCMAKE_PREFIX_PATH=/opt/gdal \
                -DUSE_PROJ=1 \
                -DWITH_PROTOBUFC=0 && \
                make && make install 
```
In version 7.6.4, although it has a warn (/usr/local/src/mapserver/textlayout.c:332:5: warning: 'hb_font_funcs_set_glyph_func' is deprecated), everything works.

In version 8.0.0, several warnings and ends with an error.


## Attach the build log
[build-mapserver-7.log](https://github.com/MapServer/MapServer/files/9974990/build-mapserver-7.log)
[build-mapserver-8.log](https://github.com/MapServer/MapServer/files/9974991/build-mapserver-8.log)

## Operating system
Is a Docker image debian:stretch

## MapServer version and installation method
The 8.0.0 version from compiling source

--- a/src/mapgml.c
+++ b/src/mapgml.c
@@ -2166,6 +2166,13 @@ int msGMLWriteWFSQuery(mapObj *map, FILE *stream,
   return (MS_SUCCESS);
 
 #else  /* Stub for mapscript */
+  (void)map;
+  (void)stream;
+  (void)default_namespace_prefix;
+  (void)outputformat;
+  (void)nWFSVersion;
+  (void)bUseURN;
+  (void)bGetPropertyValueRequest;
   msSetError(MS_MISCERR, "WFS server support not enabled",
              "msGMLWriteWFSQuery()");
   return MS_FAILURE;

--- a/src/mapogcapi.cpp
+++ b/src/mapogcapi.cpp
@@ -126,7 +126,7 @@ static void outputError(OGCAPIErrorType errorType,
 
 static int includeLayer(mapObj *map, layerObj *layer) {
   if (!msOWSRequestIsEnabled(map, layer, "AO", "OGCAPI", MS_FALSE) ||
-      !msWFSIsLayerSupported(layer) || !msIsLayerQueryable(layer)) {
+      !msIsLayerSupportedForWFSOrOAPIF(layer) || !msIsLayerQueryable(layer)) {
     return MS_FALSE;
   } else {
     return MS_TRUE;

--- a/src/mapows.h
+++ b/src/mapows.h
@@ -547,7 +547,7 @@ wfsParamsObj *msWFSCreateParamsObj(void);
 int msWFSHandleUpdateSequence(mapObj *map, wfsParamsObj *wfsparams,
                               const char *pszFunction);
 void msWFSFreeParamsObj(wfsParamsObj *wfsparams);
-int msWFSIsLayerSupported(layerObj *lp);
+int msIsLayerSupportedForWFSOrOAPIF(layerObj *lp);
 int msWFSException(mapObj *map, const char *locator, const char *code,
                    const char *version);
 

--- a/src/mapwfs.cpp
+++ b/src/mapwfs.cpp
@@ -514,26 +514,8 @@ static int msWFSGetFeatureApplySRS(mapObj *map, const char *srs,
   return MS_SUCCESS;
 }
 
-/* msWFSIsLayerSupported()
-**
-** Returns true (1) is this layer meets the requirements to be served as
-** a WFS feature type.
-*/
-int msWFSIsLayerSupported(layerObj *lp) {
-  /* In order to be supported, lp->type must be specified, even for
-  ** layers that are OGR, SDE, SDO, etc connections.
-  */
-  if ((lp->type == MS_LAYER_POINT || lp->type == MS_LAYER_LINE ||
-       lp->type == MS_LAYER_POLYGON) &&
-      lp->connectiontype != MS_WMS && lp->connectiontype != MS_GRATICULE) {
-    return 1; /* true */
-  }
-
-  return 0; /* false */
-}
-
 static int msWFSIsLayerAllowed(layerObj *lp, owsRequestObj *ows_request) {
-  return msWFSIsLayerSupported(lp) &&
+  return msIsLayerSupportedForWFSOrOAPIF(lp) &&
          (msIntegerInArray(lp->index, ows_request->enabled_layers,
                            ows_request->numlayers));
 }
@@ -4714,6 +4696,10 @@ int msWFSDispatch(mapObj *map, cgiRequestObj *requestobj,
   return returnvalue;
 
 #else
+  (void)map;
+  (void)requestobj;
+  (void)ows_request;
+  (void)force_wfs_mode;
   msSetError(MS_WFSERR, "WFS server support is not available.",
              "msWFSDispatch()");
   return (MS_FAILURE);
@@ -5590,6 +5576,29 @@ int msWFSParseRequest(mapObj *map, cgiRequestObj *request,
     msFree(pszSchemaLocation);
   }
 
+#else
+  (void)map;
+  (void)request;
+  (void)wfsparams;
+  (void)force_wfs_mode;
 #endif
   return MS_SUCCESS;
+}
+
+/* msIsLayerSupportedForWFSOrOAPIF()
+**
+** Returns true (1) is this layer meets the requirements to be served as
+** a WFS feature type or OGC API Features.
+*/
+int msIsLayerSupportedForWFSOrOAPIF(layerObj *lp) {
+  /* In order to be supported, lp->type must be specified, even for
+  ** layers that are OGR, SDE, SDO, etc connections.
+  */
+  if ((lp->type == MS_LAYER_POINT || lp->type == MS_LAYER_LINE ||
+       lp->type == MS_LAYER_POLYGON) &&
+      lp->connectiontype != MS_WMS && lp->connectiontype != MS_GRATICULE) {
+    return 1; /* true */
+  }
+
+  return 0; /* false */
 }

--- a/src/mapwfs11.cpp
+++ b/src/mapwfs11.cpp
@@ -520,7 +520,7 @@ int msWFSGetCapabilities11(mapObj *map, wfsParamsObj *params,
                           ows_request->numlayers))
       continue;
 
-    if (msWFSIsLayerSupported(lp))
+    if (msIsLayerSupportedForWFSOrOAPIF(lp))
       xmlAddChild(psFtNode, msWFSDumpLayer11(map, lp, psNsOws, OWS_1_1_0, NULL,
                                              script_url));
   }

--- a/src/mapwfs20.c
+++ b/src/mapwfs20.c
@@ -815,7 +815,7 @@ int msWFSGetCapabilities20(mapObj *map, wfsParamsObj *params,
                           ows_request->numlayers))
       continue;
 
-    if (msWFSIsLayerSupported(lp)) {
+    if (msIsLayerSupportedForWFSOrOAPIF(lp)) {
       if (psFtNode != NULL) {
         xmlAddChild(psFtNode, msWFSDumpLayer11(map, lp, psNsOws, OWS_2_0_0,
                                                validated_language, script_url));
@@ -1137,7 +1137,7 @@ int msWFSListStoredQueries20(mapObj *map, owsRequestObj *ows_request) {
 
               if (!msIntegerInArray(lp->index, ows_request->enabled_layers,
                                     ows_request->numlayers) ||
-                  !msWFSIsLayerSupported(lp))
+                  !msIsLayerSupportedForWFSOrOAPIF(lp))
                 continue;
 
               value = msOWSLookupMetadata(&(map->web.metadata), "FO",
@@ -1372,7 +1372,7 @@ int msWFSDescribeStoredQueries20(mapObj *map, wfsParamsObj *params,
 
               if (!msIntegerInArray(lp->index, ows_request->enabled_layers,
                                     ows_request->numlayers) ||
-                  !msWFSIsLayerSupported(lp))
+                  !msIsLayerSupportedForWFSOrOAPIF(lp))
                 continue;
 
               value = msOWSLookupMetadata(&(map->web.metadata), "FO",

